pkg/tsdb/prometheus/streaming/client/client.go

```diff
@@ -0,0 +1,104 @@
+package client
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/grafana/grafana/pkg/tsdb/prometheus/query"
+)
+
+type doer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type Client struct {
+	doer    doer
+	method  string
+	baseUrl string
+}
+
+func NewClient(d doer, method, baseUrl string) *Client {
+	return &Client{doer: d, method: method, baseUrl: baseUrl}
+}
+
+func (c *Client) QueryRange(ctx context.Context, q *query.Query) (*http.Response, error) {
+	u, err := url.ParseRequestURI(c.baseUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	u.Path = path.Join(u.Path, "api/v1/query_range")
+
+	qs := u.Query()
+	qs.Set("query", q.Expr)
+	tr := q.TimeRange()
+	qs.Set("start", formatTime(tr.Start))
+	qs.Set("end", formatTime(tr.End))
+	qs.Set("step", strconv.FormatFloat(tr.Step.Seconds(), 'f', -1, 64))
+	u.RawQuery = qs.Encode()
+
+	return c.fetch(ctx, u, qs)
+}
+
+func (c *Client) QueryInstant(ctx context.Context, q *query.Query) (*http.Response, error) {
+	u, err := url.ParseRequestURI(c.baseUrl)
+	if err != nil {
+		return nil, err
+	}
+
+	u.Path = path.Join(u.Path, "api/v1/query")
```

Comment:
is this needed, or should we overwrite the path from the user supplied URL string?

Reply:
I don't know enough to say 🤷‍♂️